Src/GoogleApis.Auth.DotNet4/OAuth2/DefaultCredentialProvider.cs

```diff
@@ -0,0 +1,263 @@
+                    // Catching generic exception type because any corrupted file could manifest in different ways including 
+                    // but not limited to the System, System.IO or from the Newtonsoft.Json namespace.
+                    throw new InvalidOperationException(String.Format("Error reading credential file from location {0}: {1}"
+                                            + "\nPlease check the value of the Environment Variable {2}",
+                                            credentialPath,
+                                            e.Message,
+                                            CredentialEnvironmentVariable));
+                }
+            }
+            
+            // 2. Then try the well known file.
+            credentialPath = GetWellKnownCredentialFilePath();
+            if (!String.IsNullOrWhiteSpace(credentialPath))
+            {
+                try
+                {
+                    return CreateDefaultCredentialFromFile(credentialPath);
+                }
+                catch (FileNotFoundException)
+                {
+                    // File is not present, eat the exception and move on to the next check.
+                    Logger.Debug(String.Format("Well-known credential file {0} not found.", credentialPath));
+                }
+                catch (Exception e)
+                {
+                    throw new InvalidOperationException(String.Format("Error reading credential file from location {0}: {1}"
+                                        + "\nPlease rerun 'gcloud auth login' to regenerate credentials file.",
+                                        credentialPath,
+                                        e.Message));
+                }
+            }
+
+            // 3. Then try the compute engine.     
+            Logger.Debug("Checking whether the application is running on ComputeEngine.");
+            if (await ComputeCredential.IsRunningOnComputeEngine().ConfigureAwait(false))
+            {
+                Logger.Debug("ComputeEngine check passed. Using ComputeEngine Credentials.");
+                return GoogleCredential.FromCredential(new ComputeCredential());
+            }
+
+            // If everything we could try has failed, report an error.
+            throw new InvalidOperationException(String.Format("The Application Default Credentials are not available. They are available if running"
+                + " in Google Compute Engine. Otherwise, the environment variable {0} must be defined"
+                + " pointing to a file defining the credentials. See {1} for more information.",
+                CredentialEnvironmentVariable,
+                HelpPermalink));
+        }
+
+        /// <summary>Creates default credential from a JSON file.</summary>
+        private GoogleCredential CreateDefaultCredentialFromFile(string credentialPath)
+        {
+            Logger.Debug(String.Format("Loading Credential from file {0}", credentialPath));
```

Comment:
Yeah, thanks for noticing that. I found 2 locations where I was doing this and fixed them.